### PR TITLE
[docs] update html-minifier docs

### PR DESCRIPTION
**Pull request by benmccann**
closes https://github.com/sveltejs/kit/issues/3488

### CODE REVIEW

**Comment by Rich-Harris** [comment]
I kinda wonder if we should implement a `transformPage` option on `resolve` to handle this use case:

```js
const response = await resolve(request, {
  transformPage: ({ html }) => minify(html, minification_options)
});
```

Creating a duplicate response feels a bit wasteful

**Comment by Conduitry** [comment]
That feels like a little bit of a confusing and niche API for this. What would it get passed besides `html`? Would it only get called for `text/html` content? Would it let you do anything to modify headers or other content besides the body? This feels like a hook within a hook.

**Comment by benmccann** [comment]
It is a bit verbose currently. I also wonder if it'd be a big niche to offer a helper. I do think if we were going to offer a helper it should allow you to override more than just the body such as:
```
cloneResponse(original, { status: newStatus, body: newBody, headers: newHeaders })
```

**Comment by Rich-Harris** [comment]
It would be purely for manipulating rendered pages. This line... https://github.com/sveltejs/kit/blob/0cca67063a4f38806eb4428b76548b8d948035ca/packages/kit/src/runtime/server/page/render.js#L218 ...would become something like 

```js
const html = options.transformPage(options.template({ head, body, assets }));
```

There are other reasons you might want to manipulate the HTML, like replacing `<html lang="%lang%">` with the current language (yes, Kit should provide first-class support for that anyway, but until it does...), plus probably others we haven't thought of. So it's not _just_ scenario-solving

**Comment by Conduitry** [comment]
We should merge this PR as-is anyway. I don't want to rush into an API for a new feature because it's holding up a fix in the docs about how the system currently works.

**Comment by Rich-Harris** [comment]
I've opened #3511

File: documentation/migrating/99-integrations.md [hunk 1]
```diff
@@ -34,8 +34,11 @@ const minification_options = {
 export async function handle({ event, resolve }) {
   const response = await resolve(event);
 
-  if (prerendering && response.headers['content-type'] === 'text/html') {
-    response.body = minify(response.body, minification_options);
+  if (prerendering && response.headers.get('content-type') === 'text/html') {
+    return new Response(minify(response.text(), minification_options), {
```
Comment (Rich-Harris) [inline documentation/migrating/99-integrations.md:38]:
```suggestion
    return new Response(minify(await response.text(), minification_options), {
```

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**